source/parser.hera

```diff
@@ -3639,6 +3666,24 @@ Condition
       expression,
     }
 
+DeclarationCondition
+  LexicalDeclaration:dec ->
+    const ref = {
+      type: "Ref",
+      base: "ref",
+    }
+
+    const { binding, initializer } = dec
+
+    const initCondition = {
+      type: "AssignmentExpression",
+      children: [ref, " ", initializer],
+      hoistableDec: [["", ["let ", ref], ";\n"]],
```

Comment:
"Hoistable" suggests that it *could* be hosted but doesn't need to be. Maybe `hoistDec` would make more sense?